### ISSUE TITLE
agoric-upgrade-10 error: board cannot marshal Promise

### ISSUE THREAD

**Issue by arirubinstein** [Agoric/Agoric#7643]
```
2023-05-08T19:31:37.652Z SwingSet: xsnap: v1: UnhandledPromiseRejectionWarning: (RemoteError(error:liveSlots:v6#70001)#1)
2023-05-08T19:31:37.653Z SwingSet: xsnap: v1: RemoteError(error:liveSlots:v6#70001)#1: invalid key type for collection "valToId": A "promise" cannot be a scalar key: (an object)
2023-05-08T19:31:37.653Z SwingSet: xsnap: v1: Error: invalid key type for collection "valToId": A "promise" cannot be a scalar key: (an object)
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:281)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:434)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:356)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1225)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1249)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1514)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:60)
 at ()

2023-05-08T19:31:37.658Z SwingSet: xsnap: v1: UnhandledPromiseRejectionWarning: (RemoteError(error:liveSlots:v6#70002)#2)
2023-05-08T19:31:37.658Z SwingSet: xsnap: v1: RemoteError(error:liveSlots:v6#70002)#2: invalid key type for collection "valToId": A "promise" cannot be a scalar key: (an object)
2023-05-08T19:31:37.659Z SwingSet: xsnap: v1: Error: invalid key type for collection "valToId": A "promise" cannot be a scalar key: (an object)
 at construct ()
 at Error (/bundled-source/.../node_modules/ses/src/error/tame-error-constructor.js:56)
 at makeError (/bundled-source/.../node_modules/ses/src/error/assert.js:243)
 at decodeErrorCommon (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:281)
 at decodeFromSmallcaps (/bundled-source/.../node_modules/@endo/marshal/src/encodeToSmallcaps.js:434)
 at fromCapData (/bundled-source/.../node_modules/@endo/marshal/src/marshal.js:356)
 at notifyOnePromise (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1225)
 at notify (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1249)
 at dispatchToUserspace (/bundled-source/.../packages/swingset-liveslots/src/liveslots.js:1514)
 at runWithoutMetering (/bundled-source/.../packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js:60)
 at ()

```

**Comment by warner** [comment]
The fatal message was a `write()` to v7-board, whose arguments included an array of `[name, value]` pairs, where more than half the values were Promises:

| name              | kref  | iface                         |
| ---------------- | ----- | ----------------------------- |
| centralSupply     | ko80  | Alleged: BundleIDInstallation |
| contractGovernor  | ko136 | Alleged: BundleIDInstallation |
| mintHolder        | ko81  | Alleged: BundleIDInstallation |
| provisionPool     | ko134 | Alleged: BundleIDInstallation |
| walletFactory     | kp135 | Alleged: BundleIDInstallation |
| committee         | kp332 |                               |
| binaryVoteCounter | kp333 |                               |
| VaultFactory      | kp334 |                               |
| auctioneer        | kp335 |                               |
| feeDistributor    | kp336 |                               |
| reserve           | kp337 |                               |

**Comment by raphdev** [comment]
In case it's helpful, this is how I was able to reproduce: https://github.com/Agoric/agoric-sdk/pull/7644#pullrequestreview-1417550851

**Comment by dckc** [comment]
I think this is fixed in #7645
